internal/docstore/query.go

```diff
@@ -91,6 +91,15 @@ func (q *Query) Limit(n int) *Query {
 	return q
 }
 
+// BeforeQuery takes a callback function that will be called before the Query is
+// executed to the underlying provider's query functionality. The callback takes
+// a parameter, asFunc, that converts its argument to provider-specific types.
+// See https://godoc.org/gocloud.dev#hdr-As for background information.
+func (q *Query) BeforeQuery(beforeQuery func(asFunc func(interface{}) bool) error) *Query {
```

Comment:
nit: call the arg something simple like `f` to avoid repeating "beforeQuery".

Reply:
Done.